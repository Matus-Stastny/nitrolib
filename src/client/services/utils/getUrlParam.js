// @flow strict
import queryString from "query-string";
import * as R from "ramda";

const getUrlParam = (key: string): ?string =>
  typeof window !== "undefined" && window.location.search
    ? R.propOr(null, key, queryString.parse(window.location.search))
    : null;

export default getUrlParam;
