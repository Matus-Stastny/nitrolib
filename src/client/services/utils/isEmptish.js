// @flow strict
import * as R from "ramda";

const isEmptish = R.compose(R.isEmpty, R.filter(Boolean));

export default isEmptish;
