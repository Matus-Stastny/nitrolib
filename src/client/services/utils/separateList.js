// @flow strict
import * as R from "ramda";

const separateList = R.curry((n: number, list: any[]) => {
  const len = list.length;
  const f = (_v: any, idx: number) => Math.floor((idx * n) / len);
  return R.values(R.addIndex(R.groupBy)(f, list));
});

export default separateList;
