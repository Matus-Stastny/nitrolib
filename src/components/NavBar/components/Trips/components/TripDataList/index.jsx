// @flow
import * as React from "react";
import { graphql, QueryRenderer } from "react-relay";
import type { Environment } from "react-relay";
import Alert from "@kiwicom/orbit-components/lib/Alert";
import styled from "styled-components";

import Text from "../../../../../Text";
import TripHeader from "../TripHeader";
import TripList from "../TripList";
import TripContainer from "../../../../../TripsContainer";

type Props = {|
  onSelect: (bid: string) => void,
  // DI
  env: Environment,
|};

const StateContainer = styled.div`
  padding: 10px;
`;

const TripDataList = ({ env, onSelect }: Props) => (
  <QueryRenderer
    environment={env}
    query={graphql`
      query TripDataListQuery {
        customerBookings {
          ...TripList_list
          ...TripHeader_trips
        }
      }
    `}
    variables={{}}
    render={res => {
      if (res.error) {
        return (
          <TripContainer positionMenuTablet={0} positionMenuDesktop={50}>
            <StateContainer>
              <Alert type="critical">{String(res.error)}</Alert>
            </StateContainer>
          </TripContainer>
        );
      }

      if (!res.props) {
        return (
          <TripContainer positionMenuTablet={0} positionMenuDesktop={50}>
            <StateContainer>
              <Text t={__("common.loading")} />
            </StateContainer>
          </TripContainer>
        );
      }

      const { customerBookings } = res.props;
      if (!customerBookings) {
        return (
          <TripContainer positionMenuTablet={0} positionMenuDesktop={50}>
            <StateContainer>
              <Alert>
                <Text t={__("account.no_trips")} />
              </Alert>
            </StateContainer>
          </TripContainer>
        );
      }

      return (
        <TripContainer
          header={<TripHeader trips={customerBookings} />}
          positionMenuTablet={0}
          positionMenuDesktop={50}
        >
          <TripList list={customerBookings} onSelect={onSelect} />
        </TripContainer>
      );
    }}
  />
);

export default TripDataList;
