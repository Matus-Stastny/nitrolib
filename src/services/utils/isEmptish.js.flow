// @flow strict
declare export default function isEmptish(input: {}): boolean;
