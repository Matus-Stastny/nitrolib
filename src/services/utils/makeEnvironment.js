// @flow strict
import { Environment, Network, RecordSource, Store } from "relay-runtime";

export type Input = {
  query: string,
  variables: { [key: string]: string },
};

export type Call = (input: Input) => Promise<any>;

const makeFetchQuery = (call: Call) => (operation, variables) =>
  call({
    query: operation.text,
    variables,
  });

const store = new Store(new RecordSource());

const makeEnvironment = (call: Call) =>
  new Environment({
    network: Network.create(makeFetchQuery(call)),
    store,
  });

export default makeEnvironment;
