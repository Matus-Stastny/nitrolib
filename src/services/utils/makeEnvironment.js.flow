// @flow strict
import type { Environment } from "relay-runtime";

export type Input = {
  query: string,
  variables: { [key: string]: string },
};

export type Call = (input: Input) => Promise<any>;

declare export default function makeEnvironment(call: Call): Environment;
