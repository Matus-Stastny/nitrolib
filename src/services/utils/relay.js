// @flow strict
import { Environment, GraphQLTaggedNode, Network, RecordSource, Store } from "relay-runtime";
import { commitMutation } from "react-relay";

export type Input = {
  query: string,
  variables: { [key: string]: string },
};

export type Call = (input: Input) => Promise<$FlowFixMe>;

const makeFetchQuery = (call: Call) => (operation, variables) =>
  call({
    query: operation.text,
    variables,
  });

const store = new Store(new RecordSource());

export const makeCall = (token: string, kwAuthToken: string = "") => (input: Input) =>
  fetch("https://graphql.kiwi.com", {
    method: "POST",
    headers: {
      "Content-type": "application/json",
      Accept: "application/json",
      Authorization: token,
      "KW-Auth-Token": kwAuthToken,
      "X-Client": "nitro",
    },
    body: JSON.stringify(input),
  }).then(res => res.json());

export const makeEnvironment = (call: Call) =>
  new Environment({
    network: Network.create(makeFetchQuery(call)),
    store,
  });

export const executeMutation = <Variables, Result>(
  mutation: GraphQLTaggedNode,
  variables: Variables,
  environment: Environment = makeEnvironment(makeCall("")),
): Promise<Result> =>
  new Promise((resolve, reject) =>
    // $FlowExpected: Function with generics > default type that is too general
    commitMutation(environment, {
      mutation,
      variables,
      onCompleted: (result: Result) => resolve(result),
      onError: reject,
    }),
  );
