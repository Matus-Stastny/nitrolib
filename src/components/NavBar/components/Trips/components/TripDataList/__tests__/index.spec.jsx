// @flow strict
import * as React from "react";
import { shallow } from "enzyme";

import { makeEnvironment } from "../../../../../../../services/utils/relay";

import TripDataList from "..";

const res = {
  data: {
    customerBookings: {
      edges: [
        {
          node: {
            destinationImageUrl:
              "https://images.kiwi.com/photos/600x600/barcelona_es.grayscale.jpg",
            id: "Qm9va2luZ09uZVdheTozNzA1ODU5",
            databaseId: "Qm9va2luZ09uZVdheTozNzA1ODU5",
            isPastBooking: true,
            passengerCount: 2,
            __typename: "BookingOneWay",
            trip: {
              arrival: {
                airport: {
                  city: {
                    name: "Barcelona",
                  },
                  id: "bG9jYXRpb246QkNO",
                },
                localTime: "2017-08-03T23:45:00.000Z",
              },
              departure: {
                airport: {
                  city: {
                    name: "Prague",
                  },
                  id: "bG9jYXRpb246UFJH",
                },
                localTime: "2017-08-03T21:10:00.000Z",
              },
            },
          },
        },
      ],
    },
  },
};

describe("#TripDataList", () => {
  test("render", async () => {
    const promise = Promise.resolve(res);
    const environment = makeEnvironment(() => promise);

    const wrapper = shallow(<TripDataList env={environment} onSelect={jest.fn()} />);

    await promise;

    expect(wrapper.dive()).toMatchSnapshot();
  });
});
