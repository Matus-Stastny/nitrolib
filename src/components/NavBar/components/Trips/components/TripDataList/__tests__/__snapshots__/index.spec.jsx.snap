// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#TripDataList render 1`] = `
<TripContainer
  padding={true}
  positionMenuDesktop={50}
  positionMenuTablet={0}
>
  <TripDataList__StateContainer>
    <Translate
      html={false}
      t="common.loading"
      transform={[Function]}
      values={Object {}}
    />
  </TripDataList__StateContainer>
</TripContainer>
`;
